src/PostgrestBuilder.ts

```diff
@@ -8,12 +8,16 @@ import type {
   CheckMatchingArrayTypes,
   MergePartialResult,
   IsValidResultOverride,
+  ClientServerOptions,
 } from './types'
 import PostgrestError from './PostgrestError'
 import { ContainsNull } from './select-query-parser/types'
 
-export default abstract class PostgrestBuilder<Result, ThrowOnError extends boolean = false>
-  implements
+export default abstract class PostgrestBuilder<
+  ClientOptions extends ClientServerOptions,
```

Comment:
Won't block the PR based on this but I have a hunch that adding new type parameters at the start/middle of the parameter list will break some 3rd party lib relying on the typings

Reply:
Hum that's true, I've looked for usage on github: https://github.com/search?q=language%3ATypeScript++PostgrestBuilder%3C&type=code&p=3

Most of them seems to be forks though. Maybe we could pass this into a new major version though.

Or maybe in the changelog we can give a fix solution for compatibility:

```ts
import type { PostgrestBuilder as BasePostgrestBuilder } from '@supabase/posgrest-js';

type PostgrestBuilder<R, TE extends boolean = false> = BasePostgrestBuilder<{}, R, TE>
```

Didn't tested it but this should allow to use the `PostgrestBuilder` type as before 🤔 

I think I remember having trouble pushing this as a last argument because then, it required to always also define the `ThrowOnError` param which has default set.